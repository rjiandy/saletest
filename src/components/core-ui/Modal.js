// @flow

import React from 'react';

type Props = {
    content: ?React$Element<*>;
    isModalOpen: boolean;
    onClose: () => any;
};

export default function Modal(props: Props) {
    let {content, isModalOpen, onClose} = props;
    if (isModalOpen) {
        // $FlowFixMe
        document.body.style.overflow = "hidden" 
    } else {
        // $FlowFixMe
        document.body.style.overflow = "scroll"
    }
    return isModalOpen ? (
        <div style={styles.overlay} onClick={onClose}>
            <div style={styles.content}>
                {content}
            </div>
        </div>
    ) : null;
}

const styles = {
    overlay: {
        position: 'fixed',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
        backgroundColor: 'rgba(0, 0, 0, 0.8)',
        overflow: 'hidden',
    },
    content: {
        position: 'fixed',
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)',
    },
}