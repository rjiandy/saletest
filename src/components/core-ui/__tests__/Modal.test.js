import React from 'react';
import Modal from '../Modal';
import renderer from 'react-test-renderer';

it('renders correctly with props', () => {
    const tree = renderer
      .create(<Modal content={<div />} isModalOpen={true} onClose={() => null}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });