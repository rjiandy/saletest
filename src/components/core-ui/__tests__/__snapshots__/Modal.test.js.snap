// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.8)",
      "bottom": 0,
      "left": 0,
      "overflow": "hidden",
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    style={
      Object {
        "left": "50%",
        "position": "fixed",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      }
    }
  >
    <div />
  </div>
</div>
`;
